openedx/features/colaraz_features/middlewares.py

```diff
@@ -36,13 +37,13 @@ def process_request(self, request):
             if not user.is_authenticated and any(is_blocked):
                 LOGGER.info('Requested path "{}" is blocked for anonymous users'.format(request.path))
                 return self._redirect_to_login(request)
-            elif user.is_authenticated:
+            # Superuser is allowed to visit all sites
+            elif user.is_authenticated and not user.is_superuser:
                 request_site_domain = self._get_request_site_domain(request)
-                user_site_domain = self._get_user_site_domain(user, request.session)
-                if not user_site_domain or not request_site_domain:
-                    LOGGER.error('User site or request site domains are not configured properly')
-                elif user_site_domain not in request_site_domain:
-                    LOGGER.error('User "{}" can only login through {}'.format(user.username, user_site_domain))
+                user_site_identifier, is_new_user = self._get_user_site_identifier(user, request.session)
+                if not is_new_user and not (user_site_identifier and request_site_domain and 
+                        request_site_domain.startswith(user_site_identifier)):
```

Comment:
```suggestion
                if not is_new_user and not user_site_identifier and request_site_domain and 
                        request_site_domain.startswith(user_site_identifier):
```
If it's there for readability purpose, you can leave it there.

Reply:
it's a `not` on the whole condition.

Reply:
```suggestion
                if not ( is_new_user and user_site_identifier and request_site_domain and 
                        request_site_domain.startswith(user_site_identifier)):
```
Then why not this ?

Reply:
Both are quite different, you have to look in more depth 😅

Reply:
oh okay .got it 😛